dissect/target/plugins/os/unix/iptables.py

```diff
@@ -0,0 +1,146 @@
+
+            if rule_path.exists():
+                with rule_path.open("r") as h_rule:
+                    first_line = h_rule.readline()
+
+                if PATTERN_IPTABLES_SAVE_GENERATED.match(first_line):
+                    yield rule_path
+
+    @export(record=IptablesSaveRecord)
+    def iptables(self) -> Iterator[IptablesSaveRecord]:
+        """Return iptables rules saved using iptables-save."""
+
+        tzinfo = self.target.datetime.tzinfo
+        for rule_path in self._rule_files:
+            current_program = None
+            current_version = None
+            current_ts = None
+            current_table = None
+
+            with rule_path.open("r") as h_rule:
+                for line in h_rule:
+                    line = line.strip()
+
+                    if not line:
+                        continue
+
+                    elif match := PATTERN_IPTABLES_SAVE_GENERATED.match(line):
+                        current_program, current_version, ts_string = match.groups()
+                        current_ts = datetime.strptime(ts_string, self.LOG_TIME_FORMAT).replace(
+                            tzinfo=tzinfo,
+                        )
+
+                    elif line.startswith("#") or line == "COMMIT":
+                        continue
+
+                    # Table
+                    elif line.startswith("*"):
+                        current_table = line.removeprefix("*")
+
+                    # Policy
+                    elif match := PATTERN_IPTABLES_SAVE_POLICY.match(line):
+                        policy = match.groupdict()
+                        yield IptablesSaveRecord(
+                            ts=current_ts,
+                            program=current_program,
+                            version=current_version,
+                            table=current_table,
+                            chain=policy["chain"],
+                            type="policy",
+                            rule=policy["policy"],
+                            packet_count=policy["packet_count"],
+                            byte_count=policy["byte_count"],
+                            source=rule_path,
+                        )
+
+                    # Rule
+                    elif match := PATTERN_IPTABLES_SAVE_RULE.match(line):
+                        rule = match.groupdict()
+                        yield IptablesSaveRecord(
+                            ts=current_ts,
+                            program=current_program,
+                            version=current_version,
+                            table=current_table,
+                            chain=rule["chain"],
+                            type="rule",
+                            rule=rule["rule"],
+                            packet_count=rule["packet_count"] or None,
+                            byte_count=rule["byte_count"] or None,
+                            source=rule_path,
+                        )
```

Comment:
Maybe it is nice if we have some sort of mechanism to warn us if a line cannot be parsed. This will notify us about possible edge cases...

```            else:
                     self.target.log.warn(f"Unable to parse line: {line}")
```